src/main/common/form/validators/optionalIntegerValidator.ts

```
@@ -0,0 +1,21 @@
+import {ValidatorConstraint, ValidatorConstraintInterface} from 'class-validator';
+
+/**
+ * Validates that the input value is either empty or an integer with no special characters
+ */
+@ValidatorConstraint({name: 'customInt', async: false})
+export class OptionalIntegerValidator implements ValidatorConstraintInterface {
+
+  validate(value: string) {
+    const numberPattern = /^[0-9]+$/;
```

You should trim the value string before checking for any condition to remove extra spaces

---

Spaces should throw an error, it's invalid input, so the warning informs the user of that. So I don't think we should trim user input, we should just tell them it's wrong

The acceptance criteria is that any space is erroneous.